packages/portfolio-deploy/src/contract-control.js

```diff
@@ -209,18 +219,28 @@ export const prepareContractControl = (zone, svcs) => {
         return kit.creatorFacet;
       },
 
-      /** @param {string} bundleId */
-      async upgrade(bundleId) {
-        const { name, revoked, kit } = this.state;
+      /** @param {string | {bundleId: string; privateArgsOverrides?: Partial<Parameters<SF>[1]>; }} opts */
+      async upgrade(opts) {
+        if (typeof opts === 'string') opts = { bundleId: opts };
+        const { bundleId, privateArgsOverrides = {} } = opts;
+
+        const { name, revoked, kit, initialPrivateArgs } = this.state;
         trace(name, 'upgrade', bundleId);
         !revoked || Fail`revoked`;
         if (!kit) throw Fail`${q(name)}: no StartedInstanceKit`;
-        const { privateArgs } = kit;
+        const { privateArgs: previousPrivateArgs } = kit;
+        const privateArgs = {
+          ...initialPrivateArgs,
+          ...previousPrivateArgs,
+          ...privateArgsOverrides,
+        };
         const result = await E(kit.adminFacet).upgradeContract(
           bundleId,
           privateArgs,
         );
+        const newKit = harden({ ...kit, privateArgs });
         trace(name, 'upgrade result', result);
+        this.state.kit = newKit;
```

Comment:
is there any reason to keep around `initialPrivateArgs` once they have been overridden?  Are the newKit private args available to the next upgrade?

Reply:
The `initialPrivateArgs` contain powers that we need to keep across termination, to be used with new starts (the orch powers in case of the portfolio control).